core/src/main/java/io/questdb/cutlass/http/HttpConnectionContext.java

```diff
@@ -302,12 +302,17 @@ public HttpConnectionContext of(int fd, @NotNull IODispatcher<HttpConnectionCont
         return this;
     }
 
-    public boolean rejectRequest(int code, CharSequence userMessage, CharSequence cookieName, CharSequence cookieValue) throws PeerDisconnectedException, PeerIsSlowToReadException {
-        reset();
+    public HttpRequestProcessor rejectRequest(int code, CharSequence userMessage) {
+        return rejectRequest(code, userMessage, null, null);
+    }
+
+    public HttpRequestProcessor rejectRequest(int code, CharSequence userMessage, CharSequence cookieName, CharSequence cookieValue) {
         LOG.error().$(userMessage).$(" [code=").$(code).I$();
-        simpleResponse().sendStatusWithCookie(code, userMessage, cookieName, cookieValue);
-        dispatcher.registerChannel(this, IOOperation.READ);
-        return false;
+        rejectProcessor.rejectCode = code;
+        rejectProcessor.rejectMessage = userMessage;
+        rejectProcessor.rejectCookieName = cookieName;
```

Comment:
what is the plan with the cookies?

They are always null. It might be sensible to unset the cookie that is responsible for causing 401, but we don't seem to be doing this.